src/mlpack/tests/cv_test.cpp

```diff
@@ -172,4 +187,110 @@ BOOST_AUTO_TEST_CASE(TakesNumClassesTest)
       "Value should be false");
 }
 
+BOOST_AUTO_TEST_CASE(SimpleCVAccuracyTest)
+{
+  // Using the first half of data for training and the rest for validation.
+  // The validation labels are 75% correct.
+  arma::mat data =
+    arma::mat("1 0; 2 0; 1 1; 2 1; 1 0; 2 0; 1 1; 2 1").t();
+  arma::Row<size_t> labels("0 0 1 1 0 1 1 1");
+
+  SimpleCV<LogisticRegression<>, Accuracy> cv(0.5, data, labels);
+
+  BOOST_REQUIRE_CLOSE(cv.Evaluate(), 0.75, 1e-5);
+}
+
+BOOST_AUTO_TEST_CASE(SimpleCVMSETest)
+{
+  // Using the first two points for training and remaining three for validation.
+  // See the test MSETest for more explanation.
+  arma::mat data("0 1 2 3 4");
+  arma::rowvec responses("-1 0 1 3 5");
+
+  double expectedMSE = (0 * 0 + 1 * 1 + 2 * 2) / 3.0;
+
+  SimpleCV<LinearRegression, MSE> cv(0.6, data, responses);
+
+  BOOST_REQUIRE_CLOSE(cv.Evaluate(), expectedMSE, 1e-5);
```

Comment:
Right, so correct me if I am wrong.  My understanding from what you wrote in the previous case is that it is beneficial for the case of hyper-parameter tuning to have the cross-validation class store the dataset internally, so it does not need to be split multiple times.  That could save some computation.  But there is no need to save the same computation for the `SimpleCV` class as seen by the users.  So do you think it would be better to refactor a little bit, so that the `CVBase::Evaluate()` method takes an already-set-up dataset (in whatever the form you choose is, a doubled dataset or whatever else)?  Then a constructor for the hyper-parameter tuner could assemble the dataset in the right format and store it internally (thus saving computation) but `SimpleCV` would only need to do it once for a dataset.

The reason I am focusing on this bit so much is that I think it is a lot more intuitive for a user to pass the dataset/labels/weights to the `Evaluate()` function, instead of the constructor.  This "feels" more like the rest of the mlpack interface, and also allows someone to reuse a single cross-validation object for doing cross-validation on multiple datasets.